sku.0/sys.server/compiled/game/script/gm/cmd.java

```diff
@@ -654,86 +654,149 @@ else if (faction.equalsIgnoreCase("rebel"))
             target = self;
         }
         int rank = pvpGetCurrentGcwRank(target);
-        String faction = factions.getFaction(target);
-        sendSystemMessageTestingOnly(self, "(" + target + ") " + getName(target) + "'s rank = (" + rank + ") " + getString(new string_id("faction_recruiter", factions.getRankName(rank, faction))));
+        int faction = pvpGetAlignedFaction(target);
+        String factionName = null;
+		
+		if (faction == -615855020) {
+			factionName = "Imperial";
+		} else if (faction == 370444368) {
+			factionName = "Rebel";
+		}
+		
+		if (faction != 0) {
+            sendSystemMessageTestingOnly(self, "The rank of "+getPlayerName(target)+" ("+target+") is "+rank+" ("+factionName+")");
+        } else {
+            sendSystemMessageTestingOnly(self, getPlayerName(target)+ " is not aligned with a faction.");
+        }
         return SCRIPT_CONTINUE;
+		
     }
     public int cmdSetRank(obj_id self, obj_id target, String params, float defaultTime) throws InterruptedException
     {
-        if (utils.hasScriptVar(self, "setRank.pid"))
+
+		int originalRank = pvpGetCurrentGcwRank(target);
+        int currentRank = pvpGetCurrentGcwRank(target);
+        int faction = pvpGetAlignedFaction(target);
+        String command = null;
+        String rankName = null;
+
+        if (params.contains(gm.KEYWORD_TARGET))
         {
-            int oldpid = utils.getIntScriptVar(self, "setRank.pid");
-            sui.closeSUI(self, oldpid);
-            utils.removeScriptVarTree(self, "setRank");
+            params = gm.removeKeyword(params, gm.KEYWORD_TARGET);
         }
-        dictionary d = gm.parseTarget(params, target, "SETRANK");
-        if (d == null)
+        if (params == null || params.equalsIgnoreCase(""))
         {
+            sendSystemMessageTestingOnly(self, "Syntax: /setRank -target <1 to 12>");
+            sendSystemMessageTestingOnly(self, "Warning: Player will be notified of their rank change.");
             return SCRIPT_CONTINUE;
         }
-        else if (d.isEmpty())
-        {
-        }
-        else
-        {
-            params = d.getString("params");
-            obj_id oid = d.getObjId("oid");
-            if (isIdValid(oid))
-            {
-                target = oid;
-            }
-            else
-            {
-                target = self;
-            }
-        }
-        if (!isIdValid(target) || !isPlayer(target))
-        {
-            target = self;
+
+        StringTokenizer st = new StringTokenizer(params);
+        String tempGoalRank = st.nextToken();
+        int goalRank = Integer.parseInt(tempGoalRank);
+
+        if (goalRank < 1 || goalRank > 12) {
+            sendSystemMessageTestingOnly(self, "Rank must be an integer between 1 (Private) and 12 (General).");
+            return SCRIPT_CONTINUE;
         }
-        String faction = factions.getFaction(target);

```

Comment:
Why remove this block and just replace with a system message?

Reply:
It's just the diff check making it look weird. I got rid of the entire SUI window. The original command populated an SUI window where you could select the target rank (but didn't pull the correct strings) and it couldn't correctly assign a rank after you used it. I just replaced it entirely with /setRank -target <1 to 12> so I removed the SUI stuff and added handling for the singular parameter of 1 to 12 being provided instead.